### PR TITLE
Add gzip middleware to frontend HTTP endpoints

### PR DESCRIPTION
**Pull request by zalegrala**
**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #891

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -70,6 +70,7 @@
 * [ENHANCEMENT] Allow search disablement in vulture [#1069](https://github.com/grafana/tempo/pull/1069) (@zalegrala)
 * [ENHANCEMENT] Jsonnet: add `$._config.search_enabled`, correctly set `http_api_prefix` in config [#1072](https://github.com/grafana/tempo/pull/1072) (@kvrhdn)
 * [ENHANCEMENT] Performance: Remove WAL contention between ingest and searches [#1076](https://github.com/grafana/tempo/pull/1076) (@mdisibio)
+* [ENHANCEMENT] Add middleware to compress frontend HTTP responses with gzip if requested [#1080](https://github.com/grafana/tempo/pull/1080) (@kvrhdn, @zalegrala)
```
Comment (kvrhdn) [inline CHANGELOG.md:73]: Can you merge the latest changes from `main` and fix this changelog entry?

**Review by kvrhdn — COMMENTED** [review]
LGTM :+1: Already tested this earlier.

Could you also take a look at my earlier comment? I _think_ we can avoid adding `(*client).QueryTraceWithResponse`.

File: integration/e2e/compression_test.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	queryAndAssertTrace(t, apiClient, info)
+	queryAndAssertTraceCompression(t, apiClientWithCompression, info)
+}
+
+func queryAndAssertTraceCompression(t *testing.T, client *tempoUtil.Client, info *tempoUtil.TraceInfo) {
+
+	// The received client will strip the header before we have a chance to inspect it, so just validate that the compressed client works as expected.
+	result, resp, err := client.QueryTraceWithResponse(info.HexID())
+	require.NoError(t, err)
+	require.NotNil(t, resp)
```
[the remark this answers is not in the document]
Comment (zalegrala) [reply integration/e2e/compression_test.go:51]: I don't think we need it.  While I was testing, I was checking the response to determine the headers and such, but the middleware strips the headers so we don't see the content anyway.  I'll clean that up.

**Comment by zalegrala** [comment]
I've pulled in the changes from main.  Let me know if anything else here needs to be addressed.

File: CHANGELOG.md [hunk 3]
```diff
@@ -78,6 +78,7 @@
 * [ENHANCEMENT] Include tempo-cli in the release [#1086](https://github.com/grafana/tempo/pull/1086) (@zalegrala)
 * [ENHANCEMENT] Add search on span status [#1093](https://github.com/grafana/tempo/pull/1093) (@mdisibio)
 * [ENHANCEMENT] Slightly improved compression performance [#1094](https://github.com/grafana/tempo/pull/1094) (@bboreham)
+* [ENHANCEMENT] Add middleware to compress frontend HTTP responses with gzip if requested [#1080](https://github.com/grafana/tempo/pull/1080) (@zalegrala)
```
Comment (kvrhdn) [inline CHANGELOG.md:81]: Please move this into the _main / unreleased_ section. Otherwise this will be added to v1.2.0.

**Review by kvrhdn — APPROVED** [review]
LGTM 🚀  We just have to fix the changelog entry.

File: CHANGELOG.md (hunk 3 above)
Comment (zalegrala) [reply CHANGELOG.md:81]: Oh sorry, I missed that.

**Comment by kvrhdn** [comment]
CI is really upset with your latest changes, any idea what you did wrong here? 🤣 
I can run `make test-all` successfully locally...

**Comment by zalegrala** [comment]
Oh my our ci is not happy.

**Comment by zalegrala** [comment]
I've rebased on main again.

**Comment by kvrhdn** [comment]
Thanks :+1:

**Pull request merged**